datasources/consumer.go

```diff
@@ -0,0 +1,159 @@
+package datasources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/tidepool-org/clinic-worker/cdc"
+	clinics "github.com/tidepool-org/clinic/client"
+	"github.com/tidepool-org/go-common/clients"
+	"github.com/tidepool-org/go-common/events"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+)
+
+const (
+	dataSourcesTopic = "tidepool.data_sources"
+	defaultTimeout   = 30 * time.Second
+)
+
+var Module = fx.Provide(fx.Annotated{
+	Group:  "consumers",
+	Target: CreateConsumerGroup,
+})
+
+type CDCConsumer struct {
+	logger *zap.SugaredLogger
+
+	clinics clinics.ClientWithResponsesInterface
+	data    clients.DataClient
+}
+
+type Params struct {
+	fx.In
+
+	Logger  *zap.SugaredLogger
+	Clinics clinics.ClientWithResponsesInterface
+	Data    clients.DataClient
+}
+
+func CreateConsumerGroup(p Params) (events.EventConsumer, error) {
+	config, err := cdc.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	config.KafkaTopic = dataSourcesTopic
+
+	return events.NewFaultTolerantConsumerGroup(config, CreateConsumer(p))
+}
+
+func CreateConsumer(p Params) events.ConsumerFactory {
+	return func() (events.MessageConsumer, error) {
+		delegate, err := NewCDCConsumer(p)
+		if err != nil {
+			return nil, err
+		}
+		return cdc.NewRetryingConsumer(delegate), nil
+	}
+}
+
+func NewCDCConsumer(p Params) (events.MessageConsumer, error) {
+	return &CDCConsumer{
+		logger:  p.Logger,
+		clinics: p.Clinics,
+		data:    p.Data,
+	}, nil
+}
+
+func (p *CDCConsumer) Initialize(config *events.CloudEventsConfig) error {
+	return nil
+}
+
+func (p *CDCConsumer) HandleKafkaMessage(cm *sarama.ConsumerMessage) error {
+	if cm == nil {
+		return nil
+	}
+
+	return p.handleMessage(cm)
+}
+
+func (p *CDCConsumer) handleMessage(cm *sarama.ConsumerMessage) error {
+	p.logger.Debugw("handling kafka message", "offset", cm.Offset)
+	event := CDCEvent{
+		Offset: cm.Offset,
+	}
+
+	if err := p.unmarshalEvent(cm.Value, &event); err != nil {
+		p.logger.Warnw("unable to unmarshal message", "offset", cm.Offset, zap.Error(err))
+		return err
+	}
+
+	s, _ := json.MarshalIndent(event, "", "\t")
+	fmt.Println("event", string(s))
```

Comment:
`fmt.Println` can't be turned off and it should be used only for development purposes. If you think the raw event is useful for debugging purposes consider logging at `Debug` level.

Reply:
Oh, I accidentally left this in.  Was just meant to be temporary while debugging something